### ISSUE TITLE
Different meanings of `*`

### ISSUE THREAD

**Issue by fredrikekre** [KristofferC/ContMechTensors.jl#24]
I found some more bugs related to #23 and #22 . I will fix and add tests, and make a PR later. Related to this I have some thoughts on the use of the operator `*`.

Right now `*` can is used differently depending on the arguments. Lets say we have:

``` jl
x = rand() # Number
a = rand(Tensor{1,3}) # 1st order tensor
A = rand(Tensor{2,3}) # 2nd order tensor
AA = rand(Tensor{4,3}) # 4th order tensor
```

now

``` jl
x*a # `*` used to scale `a` with a number
A*a # `*` used as single contraction
AA*A # `*` used as double contraction
```

I would think it would be nice to only allow `*` to be used when a tensor is multiplied with a number. This would leave us with
- \* for scaling a tensor with a number
- ⋅ for single contraction (`dot`)
- ⊡ for double contraction (`dcontract`)

Thoughts?

PS. I ask cause many of the bugs I found are related to `*` and thus I wanna know what we decide before fixing them all.

**Comment by KristofferC** [comment]
I tried to make \* do what one intuitively would think it would do. Maybe it just end up being confusing. I am fine with doing what you suggest.

**Comment by fredrikekre** [comment]
Yea, it makes sense to use `*` as it is right now since that is kind of the classical use.
But we always know what type of tensor we are multiplying with so then you might as well select the "correct" operator. I think it is nice that operators have the same meaning as in my suggestion.

**Comment by fredrikekre** [comment]
Closed by #25
